go-controller/hybrid-overlay/pkg/controller/master.go

```diff
@@ -318,14 +318,14 @@ func setupHybridLRPolicySharedGw(nodeSubnets []*net.IPNet, nodeName string) erro
 		}
 
 		drIP := util.GetNodeHybridOverlayIfAddr(nodeSubnet).IP
-		matchStr := fmt.Sprintf(`inport == %s%s && %s.dst == %s`,
+		matchStr := fmt.Sprintf(`inport == \"%s%s\" && %s.dst == %s`,
```

Comment:
I believe the added `"` will cause duplicate item in the DB on upgrades as the `find` transaction below will not find anything because of this change (since the `match` is included as a filter in the `find`). Your commit mentioned: `Policy was not working because the inport has to be quoted`, does this mean the policy was throwing an error and was never added in the first place - in which case, there should not be a problem

Reply:
no the policy was added, but it jsut didnt work and ovn-controller complained. Let me see if i delete the master pod and if it respawns if it finds the original match

Reply:
So I just tested it doing this:
```
		drIP := util.GetNodeHybridOverlayIfAddr(nodeSubnet).IP
		matchStr := fmt.Sprintf(`inport == \"%s%s\" && %s.dst == %s`,
			ovntypes.RouterToSwitchPrefix, nodeName, L3Prefix, hybridCIDR)
		// Search for exact match to see if we need to update anything
		uuid, stderr, err := util.RunOVNNbctl("--columns", "_uuid", "--no-headings", "find", "logical_router_policy",
			"priority="+ovntypes.HybridOverlaySubnetPriority,
			"external_ids=name="+ovntypes.HybridSubnetPrefix+nodeName,
			"action=reroute",
			fmt.Sprintf("nexthops=\"%s\"", drIP),
			fmt.Sprintf(`match="%s"`, matchStr),
		)
		if err != nil {
			return fmt.Errorf("failed to run find logical_router_policy for '%s' for host %q "+
				"stderr: %s, error: %v", matchStr, nodeName, stderr, err)
		}
		// UUID exists with exact match, no update needed
		if len(uuid) > 0 {
			klog.Info("TROZET MATCH FOUND")  <---ONLY ADDED THIS
			continue
		}
```
After deleting ovnkube-master node:
```
I0319 17:38:41.133980      39 master.go:304] Setting up logical route policy for hybrid subnet on node: ovn-worker
I0319 17:38:41.134151      39 ovs.go:166] exec(135): /usr/bin/ovn-nbctl --timeout=15 --columns _uuid --no-headings find logical_router_policy priority=1002 external_ids=name=hybrid-subnet-ovn-worker action=reroute nexthops="10.244.1.3" match="inport == \"rtos-ovn-worker\" && ip4.dst == 8.0.0.0/16"
I0319 17:38:41.141100      39 ovs.go:169] exec(133): stdout: "3da0e420-3ada-47b8-ad44-498dfc8618c0\n"
I0319 17:38:41.141600      39 ovs.go:170] exec(133): stderr: ""
I0319 17:38:41.142059      39 ovs.go:166] exec(136): /usr/bin/ovn-nbctl --timeout=15 --data=bare --no-heading --columns=name find logical_router options:chassis!=null
I0319 17:38:41.146925      39 ovs.go:169] exec(134): stdout: "40b9d9f0-b4f7-4ebe-aeaa-a9ba9ff896ac\n"
I0319 17:38:41.148117      39 ovs.go:170] exec(134): stderr: ""
I0319 17:38:41.148175      39 master.go:337] TROZET MATCH FOUND
```

Reply:
ACK, thanks for confirming.